osu.Server.Spectator/Hubs/Metadata/MetadataHub.cs

```diff
@@ -186,14 +197,20 @@ protected override async Task CleanUpState(MetadataClientState state)
             await base.CleanUpState(state);
             await broadcastUserPresenceUpdate(state.UserId, null);
             await scoreProcessedSubscriber.UnregisterFromAllMultiplayerRoomsAsync(state.UserId);
+
+            using (var db = databaseFactory.GetInstance())
+            {
+                foreach (int friendId in await db.GetUserFriendsAsync(state.UserId))
+                    await Groups.RemoveFromGroupAsync(state.ConnectionId, FRIEND_PRESENCE_WATCHERS_GROUP(friendId));
+            }
```

Comment:
Ignoring the fact that this isn't reliable because the database friends may have changed since the original connection (imagine a user removed all their friends before quitting), I'm 99% sure we don't need to do this. We don't seem to do this anywhere else and so far it hasn't been an issue. One would *hope* signalr does this for you.

Reference:  https://learn.microsoft.com/en-us/aspnet/core/signalr/groups?view=aspnetcore-9.0

> Group membership isn't preserved when a connection reconnects. The connection needs to rejoin the group when it's re-established. It's not possible to count the members of a group, since this information isn't available if the application is scaled to multiple servers.

```suggestion
```

Reply:
Indeed unnecessary, removed.